packages/transformers/pwa-manifest/src/PWAManifestTransformer.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+// https://developer.mozilla.org/en-US/docs/Web/Manifest
+
+import {Transformer} from '@parcel/plugin';
+import type {MutableAsset} from '@parcel/types';
+
+import nullthrows from 'nullthrows';
+
+const getSrcHandler = (opt = {}) => (asset, dep) => {
+  dep.src = asset.addURLDependency(dep.src, opt);
+};
+
+const handleArray = (asset, list) => {
+  if (!Array.isArray(list)) {
+    return;
+  }
+  list.forEach(getSrcHandler().bind(null, asset));
+};
+
+const DEPS = {
+  icons: handleArray,
+  screenshots: handleArray,
+  serviceworker: getSrcHandler({
+    isEntry: true,
+    env: {context: 'service-worker'}
+  })
+};
+
+const collectDependencies = (asset: MutableAsset) => {
+  const ast = nullthrows(asset.ast);
+
+  Object.keys(DEPS).forEach(key => {
+    // $FlowFixMe
+    let node = ast[key];
```

Comment:
```
Cannot get ast[key] because:
 • property icons is missing in AST [1].
 • property screenshots is missing in AST [1].
 • property serviceworker is missing in AST [1].
```

is it right way to save `json` to `asset.ast`? (as in v1)

Reply:
The `AST` type has a wrapper around it.

I think it doesn't really make sense as an ast. I'd just combine the three methods below into only `transform` and not use `asset.ast` at all.